allennlp/data/tokenizers/word_filter.py

```diff
@@ -33,38 +33,71 @@ def filter_words(self, words: List[Token]) -> List[Token]:
         return words
 
 
+@WordFilter.register('regex')
+class RegexFilter(WordFilter):
+    """
+    Uses a list of regexes to filter
+    """
+    def __init__(self,
+                 patterns: List[str]) -> None:
+        self._patterns = patterns
+        self._joined_pattern = "|".join(self._patterns)
```

Comment:
I don't use regexes much - is it faster to pre-compile this somehow?

Reply:
according to [this](https://stackoverflow.com/questions/452104/is-it-worth-using-pythons-re-compile), there is negligible perf difference and it seems to be more of a stylistic concern. Using `re.compile` does look cleaner, so I'll just change it to that.